src/zero_on_drop.rs

```diff
@@ -0,0 +1,69 @@
+use std::{ops, mem, ptr, sync::atomic};
+
+// Holds a string and zeros it when we're done.
+pub struct ZeroOnDrop {
+    inner: Inner
+}
+
+impl ZeroOnDrop {
+    pub fn new() -> Self {
+        ZeroOnDrop {
+            inner: Inner(String::new())
+        }
+    }
+
+    pub fn into_inner(mut self) -> String {
+        mem::replace(&mut self.inner.0, String::new())
+    }
+
+    pub fn zero_pop(&mut self) {
```

Comment:
@tov What are you trying to do here? Could you explain line by line what you're doing, please ?

Reply:
Sure. It’s an attempt to overwrite the (CR)NL with 0s when removing them, since they won’t get zeroed by the `Drop` later. I’m not sure it’s necessary, and it could certainly be done less awkwardly.

The current code avoids assuming the UTF-8 length of the popped element. Instead, it remembers the length before it pops the last char of the string. Then it pushes 0s until the remembered length is restored. Finally, it pops those 0s again. (Doing this on my phone from memory, hope it’s right.)

Reply:
Thanks for explaining that to me.

For the sake of simplicity, and because I don't think this particular thing is a security risk, I'm going to merge the PR but then remove the `zero_pop()` part and keep `pop()` as before.

I hope that's alright with you.

Reply:
It’s more than alright with me. It was pretty silly code, but I figured it was worth noticing. Thanks!